app/assets/javascripts/discourse/app/components/bookmark-list.hbs

```diff
@@ -7,7 +7,51 @@
       <thead class="topic-list-header">
         {{#if this.site.desktopView}}
           <PluginOutlet @name="bookmark-list-table-header">
-            <th class="topic-list-data">{{i18n "topic.title"}}</th>
+            {{#if this.bulkSelectEnabled}}
+              <th class="bulk-select topic-list-data">
+                <FlatButton
+                  @action={{this.toggleBulkSelect}}
+                  @class="bulk-select"
+                  @icon="tasks"
+                  @title="bookmarks.bulk.toggle"
+                />
+              </th>
+            {{/if}}
+            <th class="topic-list-data">
+
+              {{#if this.bulkSelectEnabled}}
+                <span class="bulk-select-topics">
+                  {{~#if this.canDoBulkActions}}
+                    <div class="bulk-select-bookmarks-dropdown">
+                      <span class="bulk-select-bookmark-dropdown__count">
+                        {{i18n
+                          "bookmarks.bulk.selected_count"
+                          count=this.selectedCount
+                        }}
+                      </span>
+                      <BulkSelectBookmarksDropdown
+                        @bulkSelectHelper={{this.bulkSelectHelper}}
+                      />
+                    </div>
+
+                  {{/if~}}
+                  <button class="btn btn-default bulk-select-all">{{i18n
```

Comment:
For these two buttons, we can use `<DButton>` and `@action` since this is all in a regular Ember component.

Reply:
Great improvement, thank you 😄